sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_error.hpp

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "amqp_header.hpp"
+#include "amqp_properties.hpp"
+#include "amqp_value.hpp"
+#include "azure/core/amqp/dll_import_export.hpp"
+
+#include <azure/core/internal/extendable_enumeration.hpp>
+
+struct ERROR_INSTANCE_TAG;
+
+template <> struct Azure::Core::_internal::UniqueHandleHelper<ERROR_INSTANCE_TAG>
+{
+  static void FreeAmqpError(ERROR_INSTANCE_TAG* obj);
+
+  using type = Azure::Core::_internal::BasicUniqueHandle<ERROR_INSTANCE_TAG, FreeAmqpError>;
+};
+
+namespace Azure { namespace Core { namespace Amqp { namespace Models { namespace _internal {
+
+  using UniqueAmqpErrorHandle = Azure::Core::_internal::UniqueHandle<ERROR_INSTANCE_TAG>;
+
+  class AmqpErrorCondition
+      : public Azure::Core::_internal::ExtendableEnumeration<AmqpErrorCondition> {
+  public:
+    AmqpErrorCondition() : ExtendableEnumeration(""){};
+    AmqpErrorCondition(std::string const& value) : ExtendableEnumeration(value) {}
```

Comment:
if it is not explicit, then we in a way saying that it is a string, you can pass std::string without any conversions where AmqpErrorCondition is required, and it would be a breaking change to add explicit.